picamera2/picamera2.py

```diff
@@ -1434,47 +1427,71 @@ def capture_image_and_switch_back_(self, preview_config, name) -> Image:
                      partial(capture_image_and_switch_back_, self, preview_config, name)]
         return self.dispatch_functions(functions, wait, signal_function)
 
-    def start_encoder(self, encoder=None, output=None, pts=None, quality=Quality.MEDIUM) -> None:
+    def start_encoder(self, encoder=None, output=None, pts=None, quality=Quality.MEDIUM, name=None) -> None:
         """Start encoder
 
         :param encoder: Sets encoder or uses existing, defaults to None
         :type encoder: Encoder, optional
         :raises RuntimeError: No encoder set or no stream
         """
+        _encoder = None
         if encoder is not None:
-            self.encoder = encoder
+            _encoder = encoder
+        else:
+            if len(self._encoder) > 1:
+                raise RuntimeError("Multiple possible encoders, need to pass encoder")
+            elif len(self._encoder) == 1:
+                _encoder = list(self._encoder)[0]
+        if _encoder is None:
+            raise RuntimeError("No encoder specified")
         if output is not None:
             if isinstance(output, str):
                 output = FileOutput(output, pts=pts)
-            encoder.output = output
+            _encoder.output = output
         streams = self.camera_configuration()
-        if self.encoder is None:
-            raise RuntimeError("No encoder specified")
-        name = self.encode_stream_name
+        if name is None:
+            name = self.encode_stream_name
         if streams.get(name, None) is None:
             raise RuntimeError(f"Encode stream {name} was not defined")
-        self.encoder.width, self.encoder.height = streams[name]['size']
-        self.encoder.format = streams[name]['format']
-        self.encoder.stride = streams[name]['stride']
+        _encoder.name = name
+        _encoder.width, _encoder.height = streams[name]['size']
+        _encoder.format = streams[name]['format']
+        _encoder.stride = streams[name]['stride']
         # Also give the encoder a nominal framerate, which we'll peg at 30fps max
         # in case we only have a dummy value
         min_frame_duration = self.camera_ctrl_info["FrameDurationLimits"][1].min
         min_frame_duration = max(min_frame_duration, 33333)
-        self.encoder.framerate = 1000000 / min_frame_duration
+        _encoder.framerate = 1000000 / min_frame_duration
```

Comment:
Although not something that has been changed in this commit, looking at this makes me wonder whether we should only overwrite the framerate if it's not been set. It seems like if someone had a specific reason for setting a particular value, then maybe we should leave it?

Reply:
Good point, I'll make it so we only add the framerate if the property doesn't exist already, or it's None.